core/server/models/post.js

```diff
@@ -185,6 +189,52 @@ Post = ghostBookshelf.Model.extend({
         model.emitChange('deleted', Object.assign({usePreviousAttribute: true}, options));
     },
 
+    onDestroying: function onDestroyed(model) {
```

Comment:
Wondering why we react before the model was saved 🤔 
If an error occurs, you will still trigger the attached event, but that would be wrong.
IMO we should trigger that on saved/destroyed/updated (past)

Reply:
On destroyed, doesn't contain related models anymore that's why chose this point. It's not optimal, but there's no good way to catch the previous state otherwise :thinking:

Reply:
sorry had a typo. updated

Reply:
yes we do it `onSaving` atm, same reason as for `onDestorying`

Reply:
I have an idea. add+edit post runs always through a txn.

there is

`options.transacting.once('committed', (committed) => {`

Reply:
tried the approach with `transacting.once(...`, when calling `emitChange` from within it no events are being fired any more

Reply:
weird 🤔

Reply:
Sorry, we already wait that the txn must complete till we fire the events: https://github.com/TryGhost/Ghost/blob/master/core/server/models/base/index.js#L136

We can ignore that :)